src/linear_algebra/dimension.lean

```diff
@@ -285,6 +285,22 @@ end
 
 variables {R M}
 
+/-- A linearly-independent family of vectors in a module over a non-trivial ring must be finite if
+the module is Noetherian. -/
+lemma linear_independent.finite_of_is_noetherian [is_noetherian R M]
+  {v : ι → M} (hv : linear_independent R v) : finite ι :=
+begin
+  have hwf := is_noetherian_iff_well_founded.mp (by apply_instance : is_noetherian R M),
+  refine complete_lattice.well_founded.finite_of_independent hwf hv.independent (λ i contra, _),
+  apply hv.ne_zero i,
+  have : v i ∈ R ∙ v i := submodule.mem_span_singleton_self (v i),
+  rwa [contra, submodule.mem_bot] at this,
+end
+
+lemma linear_independent.set_finite_of_is_noetherian [is_noetherian R M]
+  {s : set M} (hi : linear_independent R (coe : s → M)) : s.finite :=
+⟨@fintype.of_finite _ hi.finite_of_is_noetherian⟩
```

Comment:
I think this might be what you want here: https://leanprover-community.github.io/mathlib_docs/data/finite/set.html#set.finite_of_finite

(I'll take a deeper look at this PR hopefully tomorrow to see what might be missing from the `finite` API.)

Reply:
Ah yes, I wrote this before #14673 was merged and had intended to use it when I could. I'll merge master and see if i can.

Reply:
Now done in https://github.com/leanprover-community/mathlib/pull/14714/commits/5b4494b4879b3a8b2c70833593ee3ea81f37e8e1

Reply:
LGTM. The only awkwardness seems to be the lack of a `fintype` projection from `finite` that has an explicit `finite` argument, but that's just cosmetic, saving a `@` here and there.